internal/pkg/exec/docker.go

```diff
@@ -110,6 +114,32 @@ func (c DockerCommand) Push(uri, imageTag string, additionalTags ...string) erro
 	return nil
 }
 
+// IsDockerEngineRunning will run `docker info` command to check if the docker engine is running.
+// And will return an empty string and nil error if the docker engine is running.
+func (c DockerCommand) IsDockerEngineRunning() (string, error) {
+	err := c.runner.Run("docker", []string{"info", "-f", "'{{json .}}'"}, command.Stdout(c.buf))
+	if err != nil {
+		return "", fmt.Errorf("get docker info: %w", err)
+	}
+	// Trim redundant prefix and suffix. For example: '{"ServerErrors":["Cannot connect...}'\n returns
+	// {"ServerErrors":["Cannot connect...}
+	c.buf = bytes.NewBuffer([]byte(strings.TrimSuffix(strings.TrimPrefix(strings.TrimSpace(c.buf.String()), "'"), "'")))
+	if strings.Contains(c.buf.String(), "command not found") {
+		return c.buf.String(), nil
+	}
+	type dockerEngineNotRunningMsg struct {
```

Comment:
I may be missing it, but if the errors are of other types/for other reasons, does this method just assume the Docker engine is not running?

Reply:
Looks like that's the only one when docker is installed but not running on macOS and Linux.